synapse/handlers/sync.py

```diff
@@ -227,14 +236,19 @@ def wait_for_sync_for_user(self, sync_config, since_token=None, timeout=0,
     @defer.inlineCallbacks
     def _wait_for_sync_for_user(self, sync_config, since_token, timeout,
                                 full_state):
+        if since_token is None:
+            sync_type = "initial_sync"
+        elif full_state:
+            sync_type = "full_state_sync"
+        else:
+            sync_type = "incremental_sync"
+
         context = LoggingContext.current_context()
         if context:
-            if since_token is None:
-                context.tag = "initial_sync"
-            elif full_state:
-                context.tag = "full_state_sync"
-            else:
-                context.tag = "incremental_sync"
+            context.tag = sync_type
+
+        if sync_config.filter_collection.lazy_load_members():
+            lazy_member_sync_counter.labels(sync_type).inc()
```

Comment:
I'm sort-of wondering if we should only bump this when the result is non-empty (ie, do it in `current_sync_for_user` rather than here). Not bothered though.